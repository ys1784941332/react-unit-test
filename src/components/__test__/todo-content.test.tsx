import { render } from '@testing-library/react';
import TodoContent from '../todo-content';

const TITLE = '这是一个标题';
const CONTENT = '这是一个内容';

describe('测试 TodoContent', ()=> {
  it('正确渲染 TodoContent 组件', ()=> {
    const { queryByText } = render(
      <TodoContent title={TITLE} content={CONTENT} />
    )

    const titleElement = queryByText(TITLE);
    const contentElement = queryByText(CONTENT);

    expect(titleElement).not.toBeNull();
    expect(titleElement).toBeInTheDocument();

    expect(contentElement).not.toBeNull();
    expect(contentElement).toBeInTheDocument();
  })

  it('正确匹配快照', ()=> {
    const { asFragment } = render(
      <TodoContent title={TITLE} content={CONTENT} />
    )
    expect(asFragment()).toMatchSnapshot();
  })
})

describe('测试定制对象的快照内容', ()=> {
  it('正确匹配快照', ()=> {
    const user = {
      name: 'lisi',
      age: 10,
    }

    expect(user).toMatchSnapshot({
      name: 'lisi',
      age: 10
    })
  })
})