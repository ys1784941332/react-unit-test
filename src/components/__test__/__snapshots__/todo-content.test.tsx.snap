// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 TodoContent 正确匹配快照 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered"
    style="width: 400px;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          这是一个标题
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <textarea
        class="ant-input"
        rows="4"
      >
        这是一个内容
      </textarea>
    </div>
  </div>
</DocumentFragment>
`;

exports[`测试定制对象的快照内容 正确匹配快照 1`] = `
Object {
  "age": 10,
  "name": "lisi",
}
`;
